src/include/miopen/image_transform/adjust_hue/problem_description.hpp

```diff
@@ -0,0 +1,105 @@
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#pragma once
+
+#include "miopen/miopen.h"
+#include "miopen/names.hpp"
+#include "miopen/problem_description_base.hpp"
+#include "miopen/tensor.hpp"
+
+namespace miopen {
+struct NetworkConfig;
+namespace image_transform {
+namespace adjust_hue {
+
+struct ProblemDescription : ProblemDescriptionBase
+{
+    ProblemDescription(const TensorDescriptor& inputTensorDesc_,
+                       const TensorDescriptor& outputTensorDesc_,
+                       const float hue_)
+        : inputTensorDesc(inputTensorDesc_), outputTensorDesc(outputTensorDesc_), hue(hue_)
+    {
+        // These are critical checks, should be run every time
+        if(!IsHueInRange())
+            MIOPEN_THROW("hue must be between -0.5 and 0.5");
+
+        if(!IsSameSize())
+            MIOPEN_THROW("input and output must have the same size");
+
+        if(!IsInputSizesValid())
+            MIOPEN_THROW("input sizes must be 4d");
+    }
+
+    NetworkConfig MakeNetworkConfig() const override;
+
+    const TensorDescriptor& GetInputTensorDesc() const { return inputTensorDesc; }
+    const TensorDescriptor& GetOutputTensorDesc() const { return outputTensorDesc; }
+    float GetHue() const { return hue; }
+
+    bool IsSameType() const { return inputTensorDesc.GetType() == outputTensorDesc.GetType(); }
+
+    bool IsSameSize() const
+    {
+        for(int i = 0; i < inputTensorDesc.GetLengths().size(); i++)
+        {
+            if(inputTensorDesc.GetLengths()[i] != outputTensorDesc.GetLengths()[i])
+            {
+                return false;
+            }
+        }
+
+        return true;
+    }
+
+    bool IsHueInRange() const { return hue >= -0.5f && hue <= 0.5f; }
+
+    bool IsInputSizesValid()
+    {
+        // We can only really support 4d tensors (ala. NCHW).
+        if(inputTensorDesc.GetLengths().size() != 4)
+            return false;
+
+        // This op only support colored images
+        if(inputTensorDesc.GetLengths()[1] != 3)
```

Comment:
This condition is also needed for image_adjust_saturation.
There are a few other things that need to be checked.
```
// image_adjust_saturation
MODNN_CHECK_ARGUMENT(saturation_factor >= 0.0,
                     "op=ImageAdjustSaturation, Saturation factor must be "
                     "greater than or equal to 0");
// image normalize
MODNN_CHECK_ARGUMENT(mean_tensor.is_contiguous() && std_tensor.is_contiguous(),
                     "op=ImageNormalize, mean and std tensor must be contiguous for "
                     "ImageNormalize");
```
Please refer to modnn and add them.

Reply:
I've added the necessary checks

Reply:
I made an unclear request earlier. I apologize.

```if(inputTensorDesc.GetLengths()[1] != 3)``` 
This condition is also needed for image_adjust_saturation.
[modnn/core/op/image_adjust_saturation.cpp:174](https://github.com/moreh-dev/backend/blob/8a6d54f3215495fde0cd13497d2d1578bc115bb4/modnn/core/op/image_adjust_saturation.cpp#L174)
```
  MODNN_CHECK_ARGUMENT(input.size(1) == 3,
                       "op=ImageAdjustSaturation, ImageAdjustSaturation only "
                       "supports RGB images");
```

Reply:
I've added the check to `adjust_saturation`